src/pykeen/trackers/file.py

```diff
@@ -0,0 +1,69 @@
+"""Tracking results in local files."""
+
+import logging
+import pathlib
+from typing import Any, Mapping, Optional, TextIO, Union
+
+from .base import ResultTracker
+
+__all__ = [
+    "CSVResultTracker",
+]
+
+logger = logging.getLogger(__name__)
+
+
+class CSVResultTracker(ResultTracker):
+    """
+    Tracking results to a CSV file.
+
+    Also allows monitoring experiments, e.g. by
+
+    .. code ::
+        tail -f results.txt | grep "hits_at_10"
+    """
+
+    #: The file where the results are written to.
+    file: TextIO
+
+    def __init__(self, path: Union[None, str, pathlib.Path] = None):
+        if path is None:
+            path = "results.txt"
+        self.path = pathlib.Path(path)
+        self.path.parent.mkdir(exist_ok=True, parents=True)
+        self.file = self.path.open(mode="w")
+
+    def start_run(self, run_name: Optional[str] = None) -> None:  # noqa: D102
+        logger.info(f"Logging results for {run_name} to {self.path.as_uri()}.")
+        # write header
+        self.file.writelines(["type,step,key,value\n"])
```

Comment:
Alternative

```suggestion
        print("type", "step", "key", "value", sep=',', file=self.file)
```

Reply:
I changed to https://docs.python.org/3/library/csv.html and support forwarding keyword arguments to it